sympy/functions/elementary/trigonometric.py

```diff
@@ -486,6 +486,11 @@ def _eval_is_zero(self):
         if arg.is_zero:
             return True
 
+    def _eval_is_complex(self):
+        if self.args[0].is_extended_real \
```

Comment:
Why extended real? I think that `AccumBounds(-1, 1)` should be finite but not complex.

Reply:
I'm not sure I understand. This is `sin._eval_is_complex` and is currently:
```python
    def _eval_is_complex(self):
        if self.args[0].is_extended_real \
                or self.args[0].is_complex:
            return True
```
So `sin(x).is_complex` gives True if `x` is a (finite) complex number or if x is extended real. The extra possibilities from extended real that aren't included in complex are `oo` and `-oo` and we have:
```julia
In [1]: sin(oo)                                                                                                                                               
Out[1]: <-1, 1>

In [2]: sin(-oo)                                                                                                                                              
Out[2]: <-1, 1>
```
AccumBounds is a strange object though because it has assumptions like a normal number:
```
In [3]: AccumBounds(1, 2).is_real                                                                                                                             
Out[3]: True

In [2]: AccumBounds(1, 2).is_complex                                                                                                                          
Out[2]: True
```
I don't know whether that's a good idea but that's how it currently is in master. I seem to remember messing around with this in a previous PR and I expect that some things in series would break without this.